controllers/metal3.io/hostfirmwaresettings_controller.go

```diff
@@ -196,14 +193,35 @@ func (r *HostFirmwareSettingsReconciler) updateStatus(info *rInfo, settings meta
 	if info.hfs.Status.Settings == nil {
 		info.hfs.Status.Settings = make(metal3v1alpha1.SettingsMap)
 	}
+	if info.hfs.Status.LocalData == nil {
+		info.hfs.Status.LocalData = make(metal3v1alpha1.SettingsMap)
+	}
 
-	// Update Status
+	// Update Status on changes
 	for k, v := range settings {
 		// Some vendors include encrypted password fields, don't add these
 		if strings.Contains(k, "Password") {
 			continue
 		}
-		info.hfs.Status.Settings[k] = v
+
+		// If this setting has ReadOnly only or Unique schema flags then store it separately
```

Comment:
Why treat Unique this way? Unique fields can be modified, no?

Reply:
This change came about from Zane's earlier comment that we should make it easier for a user to copy all of hosts's BIOS settings to another host to ensure they are configured the same way. By segregating the ReadOnly fields and fields that are unique to a host into the LocalData field then the Status.Settings can be easily copied to another host and no filtering would be needed.
BUT - you found a bug in that currently Unique fields are prevented from being set since only Status.Settings is checked for validity. Pushing an update with fix and test.

Reply:
I've changed it back to the original design to not have Unique and ReadOnly fields in a separate slice, it was just introducing too much complexity having these broken out separately.